front/types/chat.ts

```diff
@@ -12,11 +12,16 @@ export type ChatRetrievedDocumentType = {
   }[];
 };
 
+export type MessageFeedbackStatus = "positive" | "negative" | null;
+
+export type MessageRole = "user" | "retrieval" | "assistant" | "error";
 export type ChatMessageType = {
-  role: "user" | "retrieval" | "assistant" | "error";
+  sId: string | null;
```

Comment:
Should we enforce that this is not possible to have `null` here? I think we can right?